### PR TITLE
[REVIEW] Add support for `.dt.strftime`

### PR DESCRIPTION
**Pull request by galipremsagar**
Resolves: #5583 

This PR:
- [x] Adds support for `.dt.strftime`
- [x] Adds `DateTimeProperties` and `TimedeltaProperties` to `api.rst`

### CODE REVIEW

**Comment by galipremsagar** [comment]
rerun tests

File: python/cudf/cudf/core/series.py [hunk 1]
```diff
@@ -4346,6 +4498,96 @@ def _get_dt_field(self, field):
...
+            "%B",
+            "%U",
+            "%W",
+            "%c",
+            "%x",
+            "%X",
+            "%G",
+            "%u",
+            "%V",
+        ]
```
Comment (kkraus14) [inline python/cudf/cudf/core/series.py:4577]:
```suggestion
        not_implemented_formats = {
            "%a",
            "%A",
            "%w",
            "%b",
            "%B",
            "%U",
            "%W",
            "%c",
            "%x",
            "%X",
            "%G",
            "%u",
            "%V",
        }
```

File: python/cudf/cudf/core/series.py [hunk 2]
```diff
@@ -4346,6 +4498,96 @@ def _get_dt_field(self, field):
...
+        for d_format in not_implemented_formats:
+            if d_format in date_format:
+                raise NotImplementedError(
+                    f"{d_format} date-time format is not supported yet."
```
Comment (kkraus14) [inline python/cudf/cudf/core/series.py:4581]: Should we link to the github issue for someone to track it?

**Comment by kkraus14** [comment]
Thanks for the big doc improvements, looks great 😄

File: python/cudf/cudf/core/series.py (hunk 2 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/series.py]: Sounds like a good idea, added it to the error message.

**Pull request merged**